packages/router/src/lib/operations/prepare.ts

```diff
@@ -74,11 +78,11 @@ export const prepare = async (
   const bid = decodeAuctionBid(encodedBid);
   logger.info("Decoded bid from event", requestContext, methodContext, { bid });
 
-  const routerAddress = await wallet.getAddress();
+  const signerAddress = await wallet.getAddress();
   const recovered = recoverAuctionBid(bid, bidSignature);
-  if (recovered !== routerAddress) {
+  if (recovered !== signerAddress) {
     // cancellable error
-    throw new AuctionSignerInvalid(routerAddress, recovered, { methodContext, requestContext });
+    throw new AuctionSignerInvalid(signerAddress, recovered, { methodContext, requestContext });
   }
```

Comment:
should you also check if the bid has the right router address in it if you are using a contract router? otherwise signer could change and all previously accepted bids would fail here

Reply:
even if we check the router contract address, bids will fail at a later stage for relayers.